src/components/Slider/Slider.tsx

```diff
@@ -0,0 +1,204 @@
+  };
+
+  //handling color
+  let classNameList: string[] = ['slider'];
+  let sliderNumCList: string[] = ['slider-value__num-box'];
+  if (disabled) {
+    classNameList.push('slider-disabled');
+  }
+  if (color === 'primary') {
+    classNameList.push('slider-primary');
+    sliderNumCList.push('slidernum-primary');
+  }
+  if (color === 'secondary') {
+    classNameList.push('slider-secondary');
+    sliderNumCList.push('slidernum-secondary');
+  }
+  let inpClassNames = classNameList.join(' ');
+  let sliderNumClass = sliderNumCList.join(' ');
+
+  //handling size
+  let contClassNameList: string[] = ['slider-cont'];
+  if (!setSize) {
+    contClassNameList.push('slider-md');
+  }
+  if (orientation) {
+    if (orientation === 'vertical') contClassNameList.push('slider-vertical');
+    if (setSize === 'sm') {
+      contClassNameList.push('slider-vertical-sm');
+    }
+    if (setSize === 'md') {
+      contClassNameList.push('slider-vertical-md');
+    }
+    if (setSize === 'lg') {
+      contClassNameList.push('slider-vertical-lg');
+    }
+  }
+  if (setSize === 'sm') {
+    contClassNameList.push('slider-sm');
+  }
+  if (setSize === 'md') {
+    contClassNameList.push('slider-md');
+  }
+  if (setSize === 'lg') {
+    contClassNameList.push('slider-lg');
+  }
+  let contClassName = contClassNameList.join(' ');
+
+  if (range) {
+    return (
+      <div className={contClassName}>
+        <div className="slider-value-cont">
+          <div className={sliderNumClass} ref={bubbleRef}>
+            <div className="slider-value__num">
+              {sliderValue != null ? sliderValue : ''}
+            </div>
+          </div>
+        </div>
+        <input
+          type="range"
+          value={sliderValue}
+          className={inpClassNames}
+          ref={ref}
+          defaultValue={defaultValue}
+          min={range[0]}
+          max={range[range.length - 1]}
+          step={step}
+          disabled={disabled}
+          onChange={(e) => showChange(e)}
+        />
+        {
+          <div className="tickmarks">
+            {!tickMarks
+              ? ''
+              : tickMarks.map((item, index) => {
+                  return <div className="range__tick">{item}</div>;
+                })}
+          </div>
+        }
+      </div>
+    );
+  } else {
+    return (
+      <div className={contClassName}>
+        <div className="slider-value-cont">
+          <div className={sliderNumClass} ref={bubbleRef}>
+            <div className="slider-value__num">
+              {sliderValue != null ? sliderValue : ''}
+            </div>
+          </div>
+        </div>
+        <input
+          type="range"
+          value={sliderValue}
+          className={inpClassNames}
+          ref={ref}
+          defaultValue={defaultValue}
+          min={min}
+          max={max}
+          step={step}
+          disabled={disabled}
+          onChange={(e) => showChange(e)}
+        />
+      </div>
+    );
+  }
+};
```

Comment:
A lot of duplicate code here. Violates DRY principle. Could probably just hide or show the tickmarks div based on if range is true or not.